public/app/plugins/datasource/loki/result_transformer.test.ts

```diff
@@ -23,35 +23,91 @@ const streams: LokiLegacyStreamResult[] = [
   },
 ];
 
-describe('logStreamToDataFrame', () => {
-  it('converts streams to series', () => {
-    const data = streams.map(stream => legacyLogStreamToDataFrame(stream));
-
-    expect(data.length).toBe(2);
-    expect(data[0].fields[1].labels['foo']).toEqual('bar');
-    expect(data[0].fields[0].values.get(0)).toEqual(streams[0].entries[0].ts);
-    expect(data[0].fields[1].values.get(0)).toEqual(streams[0].entries[0].line);
-    expect(data[0].fields[2].values.get(0)).toEqual('1970-01-01T00:00:00Z_{foo="bar"}');
-    expect(data[1].fields[0].values.get(0)).toEqual(streams[1].entries[0].ts);
-    expect(data[1].fields[1].values.get(0)).toEqual(streams[1].entries[0].line);
-    expect(data[1].fields[2].values.get(0)).toEqual('1970-01-01T00:00:00Z_{bar="foo"}');
+const streamResult: LokiStreamResult[] = [
+  {
+    stream: {
+      foo: 'bar',
+    },
+    values: [['1970-01-01T00:00:00Z', "foo: [32m'bar'[39m"]],
+  },
+  {
+    stream: {
+      bar: 'foo',
+    },
+    values: [['1970-01-01T00:00:00Z', "bar: 'foo'"]],
+  },
+];
+
+describe('loki result transformer', () => {
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('legacyLogStreamToDataFrame', () => {
+    it('converts streams to series', () => {
+      const data = legacyStreamResult.map(stream => ResultTransformer.legacyLogStreamToDataFrame(stream));
+
+      expect(data.length).toBe(2);
+      expect(data[0].fields[1].labels['foo']).toEqual('bar');
+      expect(data[0].fields[0].values.get(0)).toEqual(legacyStreamResult[0].entries[0].ts);
+      expect(data[0].fields[1].values.get(0)).toEqual(legacyStreamResult[0].entries[0].line);
+      expect(data[0].fields[2].values.get(0)).toEqual('1970-01-01T00:00:00Z_{foo="bar"}');
+      expect(data[1].fields[0].values.get(0)).toEqual(legacyStreamResult[1].entries[0].ts);
+      expect(data[1].fields[1].values.get(0)).toEqual(legacyStreamResult[1].entries[0].line);
+      expect(data[1].fields[2].values.get(0)).toEqual('1970-01-01T00:00:00Z_{bar="foo"}');
+    });
   });
-});
 
-describe('appendResponseToBufferedData', () => {
-  it('appends response', () => {
-    const data = new MutableDataFrame();
-    data.addField({ name: 'ts', type: FieldType.time, config: { title: 'Time' } });
-    data.addField({ name: 'line', type: FieldType.string });
-    data.addField({ name: 'labels', type: FieldType.other });
-    data.addField({ name: 'id', type: FieldType.string });
-
-    appendLegacyResponseToBufferedData({ streams }, data);
-    expect(data.get(0)).toEqual({
-      ts: '1970-01-01T00:00:00Z',
-      line: "foo: [32m'bar'[39m",
-      labels: { foo: 'bar' },
-      id: '1970-01-01T00:00:00Z_{foo="bar"}',
+  describe('lokiLegacyStreamsToDataframes', () => {
+    it('should enhance data frames', () => {
+      jest.spyOn(ResultTransformer, 'enhanceDataFrame');
```

Comment:
Can we instead of checking if the specific function was called, check if the result of the transformation is correct if the derived fields are part of the config? We can basically reuse the existing tests and just add the config and then check if there is the new column in the dataframe with the datalinks.

Something like:
```
{derivedFields: [
{
  matcherRegex: 'trace=(\w+)',
  name: 'test',
  url: 'example.com'
}
]}
```
Should work for the config and should create new column with the name `test` in the resulting dataframe.

Reply:
Sure, will look at this now